concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -741,28 +737,47 @@ getSignedBlock spv tt = do
     return SignedBlock{..}
 
 -- |The bytes that are signed by a block signature.
-blockSignatureMessageBytes :: BlockHash -> BS.ByteString
-blockSignatureMessageBytes = Hash.hashToByteString . blockHash
-
--- |Verify that a block is correctly signed by the baker key present in the block.
--- (This does not authenticate that the key corresponds to a genuine baker.)
-verifyBlockSignature :: (BakedBlockData b, HashableTo BlockHash b) => b -> Bool
-verifyBlockSignature b =
+-- Note that this concatenates the provided genesis hash with the block hash.
+-- The same genesis hash must be provided when signing the block in order to verify the block signature.
+-- This function outputs the bytestring consisting of H(genesisBlockHash || blockHash).
+blockSignatureMessageBytes ::
+    -- |Hash of the genesis block
+    BlockHash ->
+    -- |Hash of the block
+    BlockHash ->
+    -- |The H(genesisBlockHash || blockHash) bytestring.
+    BS.ByteString
+blockSignatureMessageBytes genesisHash bHash = Hash.hashToByteString $! Hash.hashOfHashes (blockHash genesisHash) (blockHash bHash)
```

Comment:
Instead of `H(genesisBlockHash || blockHash)`, we could just sign `genesisBlockHash || blockHash`. We should revisit when we're reviewing our signature message discipline.

Reply:
Haven't done this now. But I'll create an issue for "reviewing signature message discipline" .

Reply:
In 3 signatures out of 4 you append the genesisBlockHash to the rest of the message, and in the 4th you hash them together. There is nothing wrong with that from a security point of view, but it certainly messes with my head :-D

Reply:
Yea, we should probably align this at some point. 
I'll let it be for now, but I have created an issue for it here: https://github.com/Concordium/concordium-node/issues/741